gateway/mw_go_plugin.go

```diff
@@ -108,11 +128,29 @@ func (m *GoPluginMiddleware) EnabledForSpec() bool {
 
 	// to record 2XX hits in analytics
 	m.successHandler = &SuccessHandler{BaseMiddleware: m.BaseMiddleware}
-
 	return true
 }
 
+func (m *GoPluginMiddleware) goPluginConfigFromRequest(r *http.Request) {
+	_, versionPaths, _, _ := m.Spec.Version(r)
+
+	found, perPathPerMethodGoPlugin := m.Spec.CheckSpecMatchesStatus(r, versionPaths, GoPlugin)
+	if found {
+		//fmt.Printf("%#v", m)
+		//fmt.Println()
+
+		m.handler = perPathPerMethodGoPlugin.(*GoPluginMiddleware).handler
+		m.Meta = perPathPerMethodGoPlugin.(*GoPluginMiddleware).Meta
+		m.Path = perPathPerMethodGoPlugin.(*GoPluginMiddleware).Path
+		m.SymbolName = perPathPerMethodGoPlugin.(*GoPluginMiddleware).SymbolName
+		m.logger = perPathPerMethodGoPlugin.(*GoPluginMiddleware).logger
+	}
+}
 func (m *GoPluginMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Request, conf interface{}) (err error, respCode int) {
+
+	// is there a go plugin per path - we copy the handler etc from the urlspec if we find one
+	m.goPluginConfigFromRequest(r)
```

Comment:
I'm not sure if it is correct behavior. 
If you have both API level plugin, and endpoint plugin, it will ignore global plugin with this approach. 
I would rather attach this middleware two times to the request pipeline, with some parameter. One will run for global plugins, one for per-endpoint.

Reply:
This works because they are attached at different places in the pipeline like you say. This function is a noop for per api go plugins as it simply looks for the gplugin config in the extended paths part of the API definition and then copies it over when found

Reply:
All tests for global plugins still pass for example

Reply:
Actually i see what you mean I think - if we have both per path and per api on same definition only the per path will be executed

Reply:
I'll add a unit test for it

Reply:
> Actually i see what you mean I think - if we have both per path and per api on same definition only the per path will be executed

Yes this one. 
For example in the case of JSVM and Virtual Endpoint, they work in parallel. And VP is just a separate middleware, which has set of shared functions to avoid code duplication.